### PR TITLE
[feat]: rename weights optimisers

### LINKED ISSUE #448: [Feature]: Give`coreax.metric.MMD` and `coreax.weights.MMD` unique names
### Describe the feature that you would like.*

There exists both `coreax.metric.MMD` and `coreax.weights.MMD`, but it is not clear to a newcomer that the following are not importing the same class:
```python
from coreax import MMD # coreax.metric.MMD
from coreax.weights import MMD # coreax.weights MMD
```

In the examples this requires:
```python
from coreax import MMD
from coreax.weights import MMD as MMDWeightsOptimiser
```

If the `coreax.weights.MMD` is renamed to `coreax.weights.MMDWeightsOptimiser` then any ambiguity is removed.

### Why is this feature valuable to you?*

Results in an easier to understand codebase.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Code of Conduct*

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by tc85324**
### PR Type
Closes #448 
- Code style update (formatting, local variables)
- Refactoring (no functional changes)

### Does this PR introduce a breaking change?
- Any uses of `coreax.weights.MMD` will need to be replaced with `coreax.weights.MMDWeightsOptimiser`.
- Any uses of `coreax.weights.SBQ` will ned to be replaced with `coreax.weights.SBQWeightsOptimiser`.

### Checklist before requesting a review
- [x] I have made sure that my PR is not a duplicate.
- [x] My code follows the style guidelines of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have performed a self-review of my code.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by gchqdev227 — CHANGES_REQUESTED** [review]
Overall this is a good change, but in order to fulfil our OSS obligations I would request that you add aliases for the `MMD` and `SBQ` classes. My suggestion would be to do something like:

```python
import warnings

class MMD(MMDWeightsOptimiser):

    def __init__(self, *args, **kwargs):
        warnings.warn("This class is now deprecated in favour of 'MMDWeightsOptimiser', and will be removed in version 0.3", category=DeprecationWarning)
        super().__init__(*args, **kwargs)
```
At some point we may wish to then to explicitly undocument these aliases, but for now it's not the end of the world if they make it in.

Also, please add an entry in the changlog!

**Comment by tc85324** [comment]
Added changelog and deprecation warnings as requested.

File: coreax/weights.py [hunk 1]
```diff
@@ -182,3 +183,19 @@ def solve(self, x: ArrayLike, y: ArrayLike, epsilon: float = 1e-10) -> Array:
         sol = coreax.util.solve_qp(kernel_mm, kernel_nm)
 
         return sol
+
+
+@deprecated("Renamed to SBQWeightsOptimiser; will be removed in version 0.3.0")
```
Comment (gchqdev227) [inline coreax/weights.py:188]: Neat decorator use!!

**Review by gchqdev227 — APPROVED** [review]
All looks good, thanks for sorting that!

**Pull request merged**